lib/train/transports/azure.rb

```diff
@@ -38,7 +38,8 @@ def initialize(options)
         @cache[:api_call] = {}
 
         if @options[:client_secret].nil? && @options[:client_id].nil?
-          parse_credentials_file
+          values = AzureHelpers::FileCredentials.parse(@options)
+          @options.merge!(values)
```

Comment:
I'd skip the local variable and inline this if there's no conditional usage of it.